ddtrace/internal/datadog/profiling/dd_wrapper/src/exporter.cpp

```diff
@@ -347,61 +363,54 @@ Profile::insert_or_get(std::string_view sv)
 bool
 Profile::reset()
 {
-    if (!ddog_prof_Profile_reset(ddog_profile, nullptr)) {
-        errmsg = "Unable to reset profile";
+    auto res = ddog_prof_Profile_reset(ddog_profile, nullptr);
+    if (!res.ok) {
+        std::string ddog_err(ddog_Error_message(&res.err).ptr);
+        errmsg = "Could not reset profile, err: " + ddog_err;
+        ddog_Error_drop(&res.err);
         std::cout << errmsg << std::endl;
         return false;
     }
     return true;
 }
 
 bool
-Profile::start_sample(unsigned int nframes)
+Profile::start_sample()
 {
     // NB, since string_storage is a deque, `clear()` may not return all of
     // the allocated space
     strings.clear();
     string_storage.clear();
     clear_buffers();
-    this->nframes = nframes;
     return true;
 }
 
 void
 Profile::push_frame_impl(std::string_view name, std::string_view filename, uint64_t address, int64_t line)
 {
-    if (cur_frame >= lines.size() || cur_frame >= locations.size())
-        return;
-
+    static const ddog_prof_Mapping null_mapping = { 0, 0, 0, to_slice(""), to_slice("") };
     name = insert_or_get(name);
     filename = insert_or_get(filename);
 
-    lines[cur_frame] = {
-      .function =
-          {
-              .name = to_slice(name),
-              .system_name = {},
-              .filename = to_slice(filename),
-              .start_line = 0,
-          },
-      .line = line,
-  };
-
-    locations[cur_frame] = {
-        {},
+    ddog_prof_Location loc = {
+        null_mapping, // No support for mappings in Python
+        {
+          to_slice(name),
+          {}, // No support for system name in Python
+          to_slice(filename),
+          0 // We don't know the start_line in the typical case
```

Comment:
AFAIK, named (or designated) initializers (forms 3 and 4 [here](https://en.cppreference.com/w/cpp/language/aggregate_initialization)) are a compiler extension until C++20 (this is somewhat annoying because it's been in C since C11, so it can be relied on in some other contexts).  I can only generally assume C++17 for now in our CI builds, and there's also the tradeoff between using "better" standards versus what end-users can reliably get in their environments.

Personally, I like named initializers and I think it improves readability, which is why I wrote it that way the first time.

Reply:
Added back in [b27f89b](https://github.com/DataDog/dd-trace-py/pull/8203/commits/b27f89bfdf9c5c5594345438e34b6cb0af317337)